src/main/java/seedu/address/model/exam/ExamDate.java

```diff
@@ -0,0 +1,75 @@
+package seedu.address.model.exam;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import java.time.DateTimeException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+
+
+/**
+ * ExamDate class represents the date of the exam.
+ */
+public class ExamDate {
+    public static final String DATE_CONSTRAINTS =
+            "Exam Date should be in YYYY-MM-DD format and must be a valid date.";
+    public final String examDate;
+    public final String dateWithoutFormatting;
+
+
+    /**
+     * Constructs a {@code ExamDate}.
+     *
+     * @param date A valid exam date.
+     */
+    public ExamDate(String date) {
+        requireNonNull(date);
+        checkArgument(isValidDate(date), DATE_CONSTRAINTS);
+        examDate = LocalDate.parse(date).format(DateTimeFormatter.ofPattern("d MMM yyyy"));
+        dateWithoutFormatting = date;
+
+    }
+
+
+
+
+    /**
+     * Returns true if a given string is a valid date.
+     */
+    public static boolean isValidDate(String date) {
+        return date.length() >= 10 && isExistingDate(date);
```

Comment:
There is an issue here where users are allowed to key in dates that have already passed. For example, 19-10-2021 is accepted as valid input even though it shouldn't

Reply:
Okay, i'll fix that bug.